repose-aggregator/components/services/datastore-service/datastore-service-impl/src/main/java/org/openrepose/core/services/datastore/impl/distributed/CacheRequest.java

```diff
@@ -65,34 +59,16 @@ private static String getCacheKey(HttpServletRequest request) {
         return cacheKey;
     }
 
-    private static String getHostKey(HttpServletRequest request) {
-        final String hostKeyHeader = request.getHeader(DatastoreHeader.HOST_KEY);
-
-        if (StringUtilities.isBlank(hostKeyHeader)) {
-            throw new MalformedCacheRequestException(MalformedCacheRequestError.NO_DD_HOST_KEY);
-        }
-
-        return hostKeyHeader;
-    }
-
     public static boolean isCacheRequestValid(HttpServletRequest request) {
         return request.getRequestURI().startsWith(CACHE_URI_PATH);
     }
 
     public static String urlFor(InetSocketAddress remoteEndpoint, String key, boolean useHttps) {
-        String proto = "http";
-        if (useHttps) {
-            proto = "https";
-        }
-        return new StringBuilder(proto).append("://").append(remoteEndpoint.getAddress().getHostAddress()).append(":").append(remoteEndpoint.getPort()).append(CACHE_URI_PATH).append(key).toString();
+        return "http" + (useHttps ? "s" : "") + "://" + remoteEndpoint.getAddress().getHostAddress() + ":" + remoteEndpoint.getPort() + CACHE_URI_PATH + key;
```

Comment:
Ah. You're totally right about this creating the same bytecode (I reread [this](http://www.pellegrino.link/2015/08/22/string-concatenation-with-java-8.html), which I think BIll linked before, and is a great read).

I would point out two things that result in relying on compiler optimization, though:

1. Changes to the compiler, or using different compilers, may result in different behavior. This is mostly a moot point because why would we ever not use `javac`? I don't even know of an alternative, actually. As for changes to the compiler, I expect that would only make things better.
2. Utilizing string concatenation here could lead future development to use string concatenation when it is inappropriate. That is, I wouldn't want a future developer (especially a more junior developer) to see/copy this without understanding why it was allowed in this context.

So yeah. I guess, in the end, it doesn't matter too much. There are pros and cons to either approach. I tend to prefer being explicit because it takes away some of the magic (which is why I'm all about static, strongly-typed languages). In this case, that magic is the compiler optimization which just results in using what is essentially a `StringBuilder` anyway. Of course, on the other hand, letting the compiler work its magic means that if the compiler gets better magic in the future, we don't have to change this.